core/src/validation.jl

```diff
@@ -23,6 +23,26 @@ isnode(sv::Type{SchemaVersion{T, N}}) where {T, N} = length(split(string(T), "."
 nodetype(sv::Type{SchemaVersion{T, N}}) where {T, N} = Symbol.(split(string(T), ".")[2:3])
 nodetype(sv::SchemaVersion{T, N}) where {T, N} = Symbol.(split(string(T), ".")[2:3])
 
+# Allowed types for downstream (to_node_id) nodes given the type of the upstream (from_node_id) node
+neighbortypes(nodetype::Symbol) = neighbortypes(Val(nodetype))
+neighbortypes(::Val{:Pump}) = Set((:Basin,))
```

Comment:
@visr Can you check these rules? And otherwise add/merge?

Reply:
Pushed in d62323e823eb0308e81aad20fa0736a0944eabb1. I think that covers it. Fun to think about.

Reply:
And fun commit message as well!